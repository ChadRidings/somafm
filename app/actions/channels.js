export const CHANNELS = 'CHANNELS';
export const FAVORITES = 'FAVORITES';

export function setHomeChannels(channel) {
  console.log('action:setHomeChannels');
  return {
    type: CHANNELS,
    home: channel
  };
}

export function setFavorites(favorites) {
  return {
    type: FAVORITES,
    favorites: favorites
  }
}
