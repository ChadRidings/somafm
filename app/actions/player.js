export const TRACK_URL = 'TRACK_URL';
export const PLAYLIST = 'PLAYLIST';
export const METADATA = 'METADATA';

export function setTrackUrl(track, playing) {
  return {
    type: TRACK_URL,
    track,
    playing
  };
}

export function setMetadata(metadata) {
  return {
    type: METADATA,
    metadata
  };
}
