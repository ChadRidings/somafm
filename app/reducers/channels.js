import { CHANNELS } from '../actions/channels';

const initialState = {
  home: [],
};

export default function track(state = initialState, action) {
  switch (action.type) {
    case CHANNELS:
      return Object.assign({}, state, {
        home: action.home
      });

    default:
      return state;
  }
}
