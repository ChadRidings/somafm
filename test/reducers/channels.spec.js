import channels from '../../app/reducers/channels';
import { CHANNELS, FAVORITES } from '../../app/actions/channels';

describe('reducers', () => {
  describe('channels', () => {
    it('should handle initial state', () => {
      expect(channels(undefined, {})).toMatchSnapshot();
    });

    // it('should handle INCREMENT_COUNTER', () => {
    //   expect(channels(1, { type: CHANNELS })).toMatchSnapshot();
    // });

    // it('should handle DECREMENT_COUNTER', () => {
    //   expect(channels(1, { type: FAVORITES })).toMatchSnapshot();
    // });

    // it('should handle unknown action type', () => {
    //   expect(channels(1, { type: 'unknown' })).toMatchSnapshot();
    // });
  });
});
