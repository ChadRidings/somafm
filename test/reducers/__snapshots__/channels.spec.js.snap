// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers channels should handle initial state 1`] = `
Object {
  "favorites": Array [],
  "home": Array [],
}
`;
